### ISSUE TITLE
No verbose mode for git-lfs pull command

### ISSUE THREAD

**Issue by lacombar** [git-lfs/git-lfs#1977]
During a pull, I want to know *which* files are being transferred, however, `git lfs pull` is lacking such a verbose mode:

Actual behavior:
```
% git lfs pull -I 'external/linux/*.deb'     
Git LFS: (12 of 12 files) 21.04 MB / 21.13 MB
```

Expected behavior:
```
% git lfs pull -I 'external/linux/*.deb'     
Downloading external/linux/file1.deb (17.59 KB)
Downloading external/linux/file2.deb (3.73 KB)
[...]
```

**Comment by technoweenie** [comment]
Thanks for the request. I can see two challenges here:

* LFS is downloading multiple files concurrently, so it may need to change the `Downloading ...` message to `Downloaded ...` when it's done.
* What about large repositories with 1000's of files?

**Comment by ttikkanen** [comment]
What kind of challenge is that? I mean the user should be able to decide how much he wants to see

**Comment by osigida** [comment]
Looks quite useful: I fetch files on CI/CD server, and logs get polluted with the fetching progress
It would be nice to have ability to disable verbose output when the build process is working fine.

**Comment by Etheryte** [comment]
Given this would be a flag that the user can enable, not the default behavior, I see no downsides to this. For regular usage, nothing would change, but for debugging this can be invaluable.

**Comment by bk2204** [comment]
I don't really have an objection to this being implemented.  We just haven't gotten to it yet.

Just FYI, for debugging, you can enable `GIT_TRACE=1` in the environment, and it will show which files are being transferred.  That may be more verbose than you want, though.

**Comment by koutheir** [comment]
`GIT_TRACE=1` is a solution, but it is too verbose. An indication of running transfers should be enough, at least informing the user that something is ongoing, instead of giving the impression that the git `pull` command is stuck.

**Comment by draeath** [comment]
Ansible might be a useful example here, as you can tell it to emit when tasks start and complete, and as it runs tasks in parallel on multiple hosts, it has to solve a similar sort of problem.

In such case, we just get a line of output when a given task/host (file download) starts, and another when it is finished (and this can include details like how long that file took, if you like).

This would be verbose for sure, but less so than `GIT_TRACE=1`

Maybe something along the lines of this?

    git-lfs: [YYYYMMDDHHMMSS.ss] started retreiving 'model-00001-of-00002.safetensors' from '<whatever>'
    git-lfs: [YYYYMMDDHHMMSS.ss] started retreiving 'model-00002-of-00002.safetensors' from '<whatever>'
    git-lfs: [YYYYMMDDHHMMSS.ss] finished retreiving 'model-00002-of-00002.safetensors' from '<whatever>' size 1459729952
    git-lfs: [YYYYMMDDHHMMSS.ss] finished retreiving 'model-00001-of-00002.safetensors' from '<whatever>' size 4965799096

Details like the timestamps, size, average rate for that file when completed, and so on could be offered with varying levels of verbosity perhaps.
